tests/test_coroutines.py

```diff
@@ -0,0 +1,80 @@
+import asyncio
+import gc
+from typing import Any
+from unittest.mock import Mock
+
+import pytest
+
+from psygnal import _async
+from psygnal._weak_callback import WeakCallback, weak_callback
+
+
+@pytest.mark.parametrize(
+    "type_",
+    [
+        "coroutinefunc",
+        "weak_coroutinefunc",
+        "coroutinemethod",
+    ],
+)
+@pytest.mark.asyncio
+async def test_slot_types(type_: str, capsys: Any) -> None:
+    backend = _async.set_async_backend("asyncio")
+    assert backend is _async.get_async_backend() is not None
+
+    await asyncio.sleep(0)
```

Comment:
```suggestion
    while True:
        if backend.running:
            break
        await asyncio.sleep(0)
```

Reply:
thanks.... yes, I'm going to have a number of best-practice questions for you :) i really don't use this stuff much.  just a moment and I will post more questions